drivers/azure/azureutil/clients.go

```diff
@@ -49,6 +52,7 @@ func (a AzureClient) resourceGroupsClient() resources.GroupsClient {
 	c.Client.UserAgent += fmt.Sprintf(";docker-machine/%s", version.Version)
 	c.RequestInspector = withInspection()
```

Comment:
@nathanleclaire hmm I think the common base on all clients is `autorest.Client`, if we were to pass it as reference, that trick might do it, let's merge this and I'll take a stab at it next.